server/plugin/api.go

```diff
@@ -1035,70 +959,223 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
 	p.writeJSON(w, info.Settings)
 }
 
-func (p *Plugin) getIssueByNumber(c *UserContext, w http.ResponseWriter, r *http.Request) {
-	owner := r.FormValue("owner")
-	repo := r.FormValue("repo")
-	number := r.FormValue("number")
-	numberInt, err := strconv.Atoi(number)
+func (p *Plugin) openAttachCommentIssueModal(c *serializer.UserContext, w http.ResponseWriter, r *http.Request) {
+	req := &serializer.OpenCreateCommentOrEditIssueModalRequestBody{}
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		c.Log.WithError(err).Warnf("Error decoding the JSON body")
+		p.writeAPIError(w, &serializer.APIErrorResponse{ID: "", Message: "Please provide a valid JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	userID := r.Header.Get(constants.HeaderMattermostUserID)
+	post, appErr := p.API.GetPost(req.PostID)
+	if appErr != nil {
+		p.writeAPIError(w, &serializer.APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s", req.PostID), StatusCode: http.StatusInternalServerError})
+		return
+	}
+	if post == nil {
+		p.writeAPIError(w, &serializer.APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s : not found", req.PostID), StatusCode: http.StatusNotFound})
+		return
+	}
+
+	p.API.PublishWebSocketEvent(
+		wsEventAttachCommentToIssue,
+		map[string]interface{}{
+			"postId": post.Id,
+			"owner":  req.RepoOwner,
+			"repo":   req.RepoName,
+			"number": req.IssueNumber,
+		},
+		&model.WebsocketBroadcast{UserId: userID},
+	)
```

Comment:
Why is the request to the plugin needed if it simply returns a websocket event?

Reply:
The same question goes for openCloseOrReopenIssueModal.

Reply:
Now we are not making any request for these cases